discovery/manager.go

```diff
@@ -41,8 +41,8 @@ import (
 )
 
 var (
-	failedConfigs = prometheus.NewCounterVec(
-		prometheus.CounterOpts{
+	failedConfigs = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
 			Name: "prometheus_sd_configs_failed_total",
```

Comment:
This should be `prometheus_sd_failed_configs` now and the help text needs to be reworded too.

Reply:
Sure, wish the rename doesn't break anything.

Reply:
We don't have stability guarantees on metrics.

Reply:
Yeah, we don't have stability guarantees on metrics, and in this case this metric changes both its type and semantic meaning, so breaking it is good in any case :) (and gauges shouldn't have a `_total` suffix according to our conventions at https://prometheus.io/docs/practices/naming/)

Reply:
No problem, will do it.